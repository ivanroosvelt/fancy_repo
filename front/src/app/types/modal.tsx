export interface ModalData {
  visible: boolean;
  title?: string;
  content?: string;
}

export interface InfoModalProps {
  onClose: () => void;
  modalData: ModalData;
}
