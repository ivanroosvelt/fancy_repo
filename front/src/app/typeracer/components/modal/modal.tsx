import { InfoModalProps } from "@/app/types/modal";


export function InfoModal(props: InfoModalProps) {
  const {
    onClose,
    modalData: { visible, title, content }
  } = props;

  return (
    visible && (
      <>
        <dialog className="fixed left-0 top-0 w-full h-full bg-black bg-opacity-50 z-50 overflow-auto backdrop-blur flex justify-center items-center">
          <div className="bg-white m-auto p-8">
            <div className="flex flex-col items-center">
              <div className="text-2xl">{title}</div>
              <div>{content}</div>
              <br />
              <button
                onClick={onClose}
                type="button"
                className="bg-red-500 text-white p-2 "
              >
                Continue
              </button>
            </div>
          </div>
        </dialog>
      </>
    )
  );
}
