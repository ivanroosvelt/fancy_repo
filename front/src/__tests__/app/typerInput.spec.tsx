import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import { InfoModal } from '../../app/typeracer/components/modal/modal';

describe('typeracer', () => {
  it('should render modal', () => {
    const inputElement = render(
      <InfoModal
        onClose={function (): void {
          throw new Error('Function not implemented.');
        }}
        modalData={{
          visible: true,
          title: 'title',
          content: 'content'
        }}
      />
    );
    expect(inputElement).toMatchSnapshot();
  });
});
