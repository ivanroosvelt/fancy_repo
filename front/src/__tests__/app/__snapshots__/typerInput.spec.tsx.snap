// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`typerInput > should render typerInput 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <dialog
        class="fixed left-0 top-0 w-full h-full bg-black bg-opacity-50 z-50 overflow-auto backdrop-blur flex justify-center items-center"
      >
        <div
          class="bg-white m-auto p-8"
        >
          <div
            class="flex flex-col items-center"
          >
            <div
              class="text-2xl"
            >
              title
            </div>
            <div>
              content
            </div>
            <br />
            <button
              class="bg-red-500 text-white p-2 "
              type="button"
            >
              Continue
            </button>
          </div>
        </div>
      </dialog>
    </div>
  </body>,
  "container": <div>
    <dialog
      class="fixed left-0 top-0 w-full h-full bg-black bg-opacity-50 z-50 overflow-auto backdrop-blur flex justify-center items-center"
    >
      <div
        class="bg-white m-auto p-8"
      >
        <div
          class="flex flex-col items-center"
        >
          <div
            class="text-2xl"
          >
            title
          </div>
          <div>
            content
          </div>
          <br />
          <button
            class="bg-red-500 text-white p-2 "
            type="button"
          >
            Continue
          </button>
        </div>
      </div>
    </dialog>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
